### PR TITLE
Multiple result sets initial support

### PR DESCRIPTION
**Pull request by kerams**
A stab at implementing #39. I haven't tried other result types besides Record yet, but the following simple example compiles and runs successfully:

```fsharp
let [<EntryPoint>] main _ =
    use cmd = DvdRental.CreateCommand<"SELECT * from film;  SELECT 'blah' AS aaa, 33 as num">(dvdRental)
    
    let res = cmd.Execute()
    let a = res.ResultSet2.Head.aaa
    let b = res.ResultSet1.Head.title
    0
```

### CODE REVIEW

**Comment by daz10000** [comment]
Nice! If you can get reasonable test coverage  would be happy to merge 

Von meinem iPhone gesendet

**Comment by kerams** [comment]
https://github.com/demetrixbio/FSharp.Data.Npgsql/blob/master/src/Runtime/ISqlCommand.fs#L35-L39

@daz10000 @melanore Do you think it would be OK to remove the SingleResult flag? I'm not sure what its purpose is besides getting in my way in this PR :).

**Comment by daz10000** [comment]
The singleresult feature is super helpful. When your query only gets a single row and you want just a record returned rather than a series of rows.  Saves everyone a lot of trouble. We use it all over our Codebase 

Von meinem iPhone gesendet

**Comment by kerams** [comment]
I think you're talking about SingleRow.

**Comment by kerams** [comment]
Ok, I think despite the following issues, this initial version could go in.

- If a command contains nonqueries only, the result is the number of affected rows for the first of them (blocked by https://github.com/npgsql/npgsql/issues/2825)
- If a command with multiple queries contains nonqueries too, no result set with the number of affected rows is generated for the latter (blocked by https://github.com/npgsql/npgsql/issues/2825)
- When using `SingleRow = true`, all result sets are considered to contain a single row. Not sure what can realistically be done about this.
- Output parameters are not implemented. I just didn't bother thinking about them for now.

**Comment by kerams** [comment]
OK, I've come up with a workaround for the first two points above, but it's not pretty :).

File: src/DesignTime/QuotationsFactory.fs [hunk 1]
```diff
@@ -660,24 +660,14 @@ type internal QuotationsFactory private() =
...
                     let propName = sprintf "ResultSet%d" (i + 1)
-                    let prop = ProvidedProperty(propName, rt.Single, fun args -> <@@ %%args.[0] |> unbox |> Map.find<string, obj> propName @@>)
+                    let prop = ProvidedProperty(propName, rt.Single, fun args -> <@@ (%%args.[0] |> unbox<obj[]>).[i] @@>)
                     let ctorParam = ProvidedParameter(propName, rt.Single)
                     prop, ctorParam)
                 |> List.unzip
 
             resultSetsType.AddMembers props
 
-            let invokeCode args =
```
Comment (kerams) [inline src/DesignTime/QuotationsFactory.fs]: This technique should be usable for the record result type too. No need to bake in property names, so quotations passed around will be much smaller. At runtime, property access and "record" instantiation will be dozens of times faster while consuming a lot less memory. It should be really noticeable when querying lots of records.

**Comment by daz10000** [comment]
Nice!   I’m a big fan of reducing quotations footprint. I suspect that’s a big part of the design time load (actually know it is) so anything that reduces this is good.  

We had a big deadline this week so sorry about slow merging but really keen to test and get these in.  

Von meinem iPhone gesendet

**Comment by melanore** [comment]
Impressive work, thank you!

**Comment by kerams** [comment]
My pleasure. There's more coming up.

**Pull request merged**